tests/foreman/ui/test_computeresource.py

```diff
@@ -18,173 +21,197 @@
 class ComputeResource(UITestCase):
     """Implements Compute Resource tests in UI"""
 
-    @data(*generate_strings_list(len1=8))
-    def test_create_resource_1(self, name):
-        """@Test: Create a new libvirt Compute Resource
+    current_libvirt_url = (LIBVERT_RESOURCE_URL
+                           % conf.properties['main.server.hostname'])
+    current_docker_url = (DOCKER_RESOURCE_URL
+                          % conf.properties['main.server.hostname'])
+
+    @data(*valid_data_list())
+    def test_create_libvirt_resource_different_names(self, name):
+        """@Test: Create a new libvirt Compute Resource using different value
+        types as a name
 
         @Feature: Compute Resource - Create
 
-        @Assert: A libvirt Compute Resource is created
+        @Assert: A libvirt Compute Resource is created successfully
 
         """
-        libvirt_url = "qemu+tcp://%s:16509/system"
-        provider_type = FOREMAN_PROVIDERS['libvirt']
-        url = (libvirt_url % conf.properties['main.server.hostname'])
         with Session(self.browser) as session:
```

Comment:
Ohh, you are about that. Do you propose to initiate ```Session``` object in ```SetUpClass```? To have one session per all test cases in one class? If that is true, so, it is really bad idea as it will ruin one of main principles of test automation for WebUI (particularly, 'entry point per each test must be always known' and you will never know about it as you will never know test output for previous test)